src/main/java/org/mandas/docker/client/LogReader.java

```diff
@@ -66,15 +63,31 @@ public LogMessage nextMessage() throws IOException {
       streamId = Stream.STDOUT.id();
       frame = new byte[stream.available()];
     }
-    ByteStreams.readFully(stream, frame);
+
+    int total = readAllBytes(frame);
+    
+    if (total < frame.length) {
+      throw new EOFException("EOF before reading " + frame.length + " bytes. " + total + " read instead.");
+    }
+    
     return new LogMessage(streamId, ByteBuffer.wrap(frame));
   }
 
+  private int readAllBytes(byte[] buf) throws IOException {
+	int total = 0;
+    while (total < buf.length) {
```

Comment:
Do you need this while loop and the 3 parameter version of `read`?  Reading the description [here](https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#read(byte[])) it seems like this will be functionally equivalent to `stream.read(buf);`

Reply:
That was my initial impression as well but tests failed at the end. The difference between the `stream.read(buf)` and this while loop is that the first one could return if just a single byte is read. What we need at this point is to block until we reach the buffer size or EOF. The previous `ByteStreams.readFully` was doing this internally as well